zombies.js

```diff
@@ -79,135 +92,213 @@
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  equip(itemToEquip) {
+    let isWeapon = itemToEquip instanceof Weapon;
+    let currentWeapon = this.equipped;
+    if (!isWeapon || this._pack.indexOf(itemToEquip) === -1) {
+      return false;
+    }
+    if (currentWeapon) {
+      this._pack.push(currentWeapon);
+    }
+    this._pack.splice(this._pack.indexOf(itemToEquip), 1);
+    this.equipped = itemToEquip;
+  }
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  eat(itemToEat) {
+    let isFood = itemToEat instanceof Food;
+    if (!isFood || this._pack.indexOf(itemToEat) === -1) {
+      return false;
+    }
+    this._pack.splice(this._pack.indexOf(itemToEat), 1);
+    if (this.health + itemToEat.energy > 100) {
+      this.health = 100;
+    } else {
+      this.health = this.health + itemToEat.energy;
+    }
+  }
+
+  /**
+   * Player Class Method => useItem(item)
+   * -----------------------------
+   * Player uses an item from the pack.
+   *
+   * If the item is a weapon, the player should equip the item.
+   * If the item is food, the player should eat the item.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name useItem
+   * @param {Item/Weapon/Food} item   The item to use.
+   */
+
+  useItem(item) {
+    if (item instanceof Weapon) {
+      this.equip(item);
+    }
+    if (item instanceof Food) {
+      this.eat(item);
+    }
+  }
+
+  /**
+   * Player Class Method => equippedWith()
+   * -----------------------------
+   * Player checks their equipment.
+   *
+   * Prints the player's name and equipped weapon's name.
+   * If nothing is equipped, prints a message saying so.
+   * Also returns the equipped weapon's name or false if nothing is equipped.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equippedWith
+   * @return {string/boolean}   Weapon name or false if nothing is equipped.
+   */
+
+  equippedWith() {
+    if (this.equipped) {
+      return this.equipped.name;
```

Comment:
This is missing the message stating the player's name and the equipped weapon's name.

Reply:
It also needs to return the weapon's name